### PR TITLE
feat: Add version to startup message

### LINKED ISSUE #428: [FEATURE] Include the version in the startup message
### Requirements

Add an info level log message during the start up process that includes version, branch, and revision.

### PR DESCRIPTION
**Pull request by joaoepj**
## This PR

- adds version to flagd startup message
### Related Issues

Fixes #428

### Notes
Running flagd from main branch:
... 

2023-02-23T17:44:37.503-0300    info    file/filepath_sync.go:32        Starting filepath sync notifier {"component": "sync", "sync": "filepath"}
2023-02-23T17:44:37.503-0300    info    service/connect_service.go:110  metrics listening at 8014       {"component": "service"}
...

Running flagd from this PR branch:
... 

2023-02-23T17:47:26.504-0300    info    cmd/start.go:97 flagd version: dev (15ef000), built at: 2023-02-23T17:47:20Z   {"component": "start"}
2023-02-23T17:47:26.504-0300    info    file/filepath_sync.go:32        Starting filepath sync notifier {"component": "sync", "sync": "filepath"}
...

### CODE REVIEW

**Comment by beeme1mr** [comment]
Hey @joaoepj, thanks for the PR! Would you mind signing off on your commit? If you're unfamiliar with DCO, you can read about it [here](https://developercertificate.org/). It's a requirements from the CNCF that all commits have been signed off by the author.

To add your Signed-off-by line to every commit in this branch:


- Ensure you have a local copy of your branch by [checking out the pull request locally via command line](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/checking-out-pull-requests-locally).
- In your local branch, run: git rebase HEAD~1 --signoff
- Force push your changes to overwrite the branch: git push --force-with-lease origin include-version-in-startup-message

**Comment by beeme1mr** [comment]
The integration test failure is a known problem and should be resolved early next week. I also don't think the drop in code coverage is an issue in this case. We should be good to go once there's enough approvals and the DCO check has passed.

**Review by Kavindu-Dodan — APPROVED** [review]
Validated locally and in my view fulfill the tasks need.

**Comment by joaoepj** [comment]
> Hey @joaoepj, thanks for the PR! Would you mind signing off on your commit? If you're unfamiliar with DCO, you can read about it [here](https://developercertificate.org/). It's a requirements from the CNCF that all commits have been signed off by the author.
> 
> To add your Signed-off-by line to every commit in this branch:
> 
> * Ensure you have a local copy of your branch by [checking out the pull request locally via command line](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/checking-out-pull-requests-locally).
> * In your local branch, run: git rebase HEAD~1 --signoff
> * Force push your changes to overwrite the branch: git push --force-with-lease origin include-version-in-startup-message

Thank you @beeme1mr, the commit was signed off!

**Review by toddbaert — APPROVED** [review]
Agreed with all the above comments. We don't need coverage for this :)

I will get this merged after rebasing on the integration test fixes.

**Pull request merged**
